### PR TITLE
increasing the clarity on the vulnerability reporting process

### PR DESCRIPTION
**Pull request by chayim**
Our vulnerability reporting process is reasonably clear, but the guidelines for getting a higher value vulnerability report need some tweaking.  I felt that:

1. Vulnerability reporting really does deserve it's own top-level page, like other organizations.

2. The 1 day (to respond) timeline is far too quick - and having now viewed several other organization, I have data on that.  The result is a table containing expectations as a response - an experiment to see if others find it easier to understand (as I do).

3. We ought clarify what constitutes a higher quality vulnerability report.

Please provide comments, I'd like to turn this into a higher value page, as I think it can help us get down the path to higher value vulnerabilities.  Finally, I validated our amounts with the few organizations I found possible. Paypal and Github have maximums of $30K. This, along with our existing bounty pay outs lead me to feel that our existing amount is well positioned given our size and scale as a result.

### CODE REVIEW

File: handbook/engineering/security/reporting-vulnerabilities.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Reporting a Vulnerability
+
+If you think that you have found a security or privacy vulnerability, please email us at <a href="mailto:security@sourcegraph.com">security@sourcegraph.com</a>. We will reply to reports within 2 US business day to acknowledge that we received them, and will strive to send you regular updates on our progress until the issue is resolved. You may request an update by replying to the existing email thread. We will read, but may not respond to low quality or spammy reports (e.g. those produced by automated tooling). We welcome reports from everyone, including security researchers, developers, and customers.
```
Comment (quinnkeast) [inline handbook/engineering/security/reporting-vulnerabilities.md:3]:
```suggestion
If you think that you have found a security or privacy vulnerability, please email us at <a href="mailto:security@sourcegraph.com">security@sourcegraph.com</a>. We will reply to reports within 2 US business days to acknowledge that we received them, and will strive to send you regular updates on our progress until the issue is resolved. You may request an update by replying to the existing email thread. We will read, but may not respond to low quality or spammy reports (e.g. those produced by automated tooling). We welcome reports from everyone, including security researchers, developers, and customers. 
```

**Review by ElizabethStirling — APPROVED** [review]
Looks great! Thanks so much for hammering this out!

File: handbook/engineering/security/reporting-vulnerabilities.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Reporting a Vulnerability
```
Comment (nicksnyder) [inline handbook/engineering/security/reporting-vulnerabilities.md:1]: https://about.sourcegraph.com/handbook/communication/content_guidelines/style_and_mechanics#capitalization

```suggestion
# Reporting a vulnerability
```

File: handbook/engineering/security/reporting-vulnerabilities.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+**Timelines**
+
+| Type of Response       | Time to Response     |
+| :------------- | :----------: |
+| First Response| 2 days |
+| Time to Triage | 10 days |
+| Time to bounty determination | 20 days |
+| Time to Resolution | depends on severity and complexity |
+| Time to payment | 90 days from the original report, or after confirmation of fix, whichever is first|
```
Comment (nicksnyder) [inline handbook/engineering/security/reporting-vulnerabilities.md:19]: https://about.sourcegraph.com/handbook/communication/content_guidelines/style_and_mechanics#capitalization

```suggestion
| First response| 2 days |
| Time to triage | 10 days |
| Time to bounty determination | 20 days |
| Time to resolution | depends on severity and complexity |
| Time to payment | 90 days from the original report, or after confirmation of fix, whichever is first|
```

File: handbook/engineering/security/reporting-vulnerabilities.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+We provide monetary rewards, from \$50 to \$10,000 USD, for security vulnerability reports. The actual reward amount is determined based on the number of customers impacted, the difficulty of exploiting the vulnerability, and the severity of the consequences (e.g. service disruption, data leakage, reputational damage to Sourcegraph) of a successful exploit.
+
+We will send payment to a valid PayPal account. We will ask you for the name and country associated with your PayPal account.
+
+**Timelines**
+
+| Type of Response       | Time to Response     |
+| :------------- | :----------: |
+| First Response| 2 days |
+| Time to Triage | 10 days |
```
Comment (nicksnyder) [inline handbook/engineering/security/reporting-vulnerabilities.md:16]: I don't think it is clear to the reader what "triage" means.

File: handbook/engineering/security/reporting-vulnerabilities.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+1. You report an already reported bounty, or one already in our roadmap.
+1. You publicly disclose a vulnerability before we confirm that it is OK to do so. We want to give our customers time to upgrade to a patched version before public disclosure.
+1. You spam us with duplicate and/or low quality vulnerability reports (e.g. copy/pasting generic issues from automatic scanning tools).
+1. You are a current or former teammate at Sourcegraph (e.g. employee, contractor, intern).
+1. You are friends or family with a current or former teammate at Sourcegraph.
+1. You are a resident of, or make your Submission from, a country against which the United States has issued export sanctions or other trade restrictions (e.g., Cuba, Iran, North Korea, Sudan and Syria);
+1. You are in violation of any national, state, or local law or regulation;
+1. You are less than 14 years of age. If you are at least 14 years old, but are considered a minor in your place of residence, you must get your parent’s or legal guardian’s permission prior to participating in the program.
+
+### Submission Requirements
```
Comment (nicksnyder) [inline handbook/engineering/security/reporting-vulnerabilities.md:35]:
```suggestion
### Submission requirements
```

File: handbook/engineering/security/reporting-vulnerabilities.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+For all submissions, please include:
+
+1. A full description of the vulnerability being reported.  This includes the exploitability and impact
```
Comment (nicksnyder) [inline handbook/engineering/security/reporting-vulnerabilities.md:39]:
```suggestion
1. A full description of the vulnerability being reported.  This includes the exploitability and impact.
```

File: handbook/engineering/security/reporting-vulnerabilities.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    1. Screenshots
+    1. Videos
+    1. Traffic logs
+    1. Complete web and API requests and responses
+    1. Email address, or IP address used during testing
+
+
+## How we respond to security vulnerability reports
+
+When we receive [a report of a security vulnerability](#how-to-report-a-security-vulnerability), a member of our security team determines if a reported vulnerability should be investigated by an engineer.
```
Comment (nicksnyder) [inline handbook/engineering/security/reporting-vulnerabilities.md:51]: This self-referential link is dead.

File: handbook/engineering/security/reporting-vulnerabilities.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+If you think that you have found a security or privacy vulnerability, please email us at <a href="mailto:security@sourcegraph.com">security@sourcegraph.com</a>. We will reply to reports within 2 US business days to acknowledge that we received them, and will strive to send you regular updates on our progress until the issue is resolved. You may request an update by replying to the existing email thread. We will read, but may not respond to low quality or spammy reports (e.g. those produced by automated tooling). We welcome reports from everyone, including security researchers, developers, and customers. 
+
+### Bounties
+
+We provide monetary rewards, from \$50 to \$10,000 USD, for security vulnerability reports. The actual reward amount is determined based on the number of customers impacted, the difficulty of exploiting the vulnerability, and the severity of the consequences (e.g. service disruption, data leakage, reputational damage to Sourcegraph) of a successful exploit.
+
+We will send payment to a valid PayPal account. We will ask you for the name and country associated with your PayPal account.
+
+**Timelines**
```
Comment (nicksnyder) [inline handbook/engineering/security/reporting-vulnerabilities.md:11]: In the first section you say we respond within 2 US business days, but it is unclear in this table whether these are also all US business days.

**Review by nicksnyder — COMMENTED** [review]
Update dead link to this page: https://sourcegraph.com/search?q=r:sourcegraph/about+how+to+report&patternType=literal&utm_source=omnibox

**Review by quinnkeast — APPROVED** [review]

**Pull request merged**